.github/workflows/build-linux.yml

```diff
@@ -24,12 +24,15 @@ jobs:
       matrix:
         # Github Actions requires a single row to be added to the build matrix.
         # See https://help.github.com/en/articles/workflow-syntax-for-github-actions.
-        name: [
-          ubuntu-20.04-gcc-9,
-          ubuntu-20.04-clang-9,
-          ubuntu-22.04-gcc-12,
-          ubuntu-22.04-clang-14,
-        ]
+        name:
+          [
+            ubuntu-20.04-gcc-9,
+            ubuntu-20.04-clang-9,
+            ubuntu-22.04-gcc-12,
+            ubuntu-22.04-clang-14,
+            ubuntu-24.04-gcc-13,
```

Comment:
PS I argued in a separate PR for adding a "special case" workflow that uses c++20 and gcc14 to be future proof. That will catch things like [=].

I think we also don't need gcc-12 *and* 13. I think if we "bracket" the supported versions we should be fine, right? So, my suggesting here would be to bracket "gcc-9 -> gcc-14", and remove 12 and 13.

Same reasoning for clang: bracket from 9 -> 16.

Reply:
I'll leave the adding of C++20 and GCC14 to someone else who needs it. :)

Reply:
OK, but could you fix the flow to do the bracketing, though? It's a comment on this PR